x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,146 @@
+
+	// Timeout both the ICA kick off command and the ibc transfer message at the epoch boundary
+	strideEpochTracker, found := k.GetEpochTracker(ctx, epochstypes.STRIDE_EPOCH)
+	if !found {
+		return errorsmod.Wrapf(types.ErrEpochNotFound, epochstypes.STRIDE_EPOCH)
+	}
+	endEpochTimestamp := uint64(strideEpochTracker.NextEpochStartTime)
+
+	// Determine the host zone's stToken ibc denom
+	nativeDenom := hostZone.HostDenom
+	stIbcDenom, err := k.GetStIbcDenomOnHostZone(ctx, hostZone)
+	if err != nil {
+		return err
+	}
+
+	// If the token is an stToken, we send it to the redeem holding address to be redeemed
+	// Otherwise, we send it to the stake holding address to be liquid staked
+	var destinationHoldingAddress string
+	switch token.Denom {
+	case nativeDenom:
+		destinationHoldingAddress = hostZone.CommunityPoolStakeHoldingAddress
+	case stIbcDenom:
+		destinationHoldingAddress = hostZone.CommunityPoolRedeemHoldingAddress
+	default:
+		return fmt.Errorf("Invalid community pool transfer denom: %s", token.Denom)
+	}
+
+	memo := ""
+	var msgs []proto.Message
+	msgs = append(msgs, transfertypes.NewMsgTransfer(
+		transfertypes.PortID,
+		counterpartyChannelId, // for transfers of communityPoolHostZone -> Stride
+		token,
+		hostZone.CommunityPoolDepositIcaAddress, // ICA controlled address on community pool zone
+		destinationHoldingAddress,               // Stride address, unique to each community pool / hostzone
+		clienttypes.Height{},
+		endEpochTimestamp,
+		memo,
+	))
+
+	// No need to build ICA callback data or input an ICA callback method since the callback Stride can see is only
+	//  the ICA callback, not the actual transfer callback. The transfer ack returns to the hostZone chain not Stride
+	icaCallbackId := ""
+	var icaCallbackData []byte
+
+	// Send the transaction through SubmitTx to kick off ICA commands -- no ICA callback method name, or callback args needed
+	_, err = k.SubmitTxs(ctx,
+		hostZone.ConnectionId,
+		msgs,
+		types.ICAAccountType_COMMUNITY_POOL_DEPOSIT,
+		endEpochTimestamp,
+		icaCallbackId,
+		icaCallbackData)
+	if err != nil {
+		return errorsmod.Wrapf(err, "Failed to SubmitTxs, Messages: %v, err: %s", msgs, err.Error())
+	}
+	k.Logger(ctx).Info("Successfully sent ICA command to kick off ibc transfer from deposit ICA to stake holding address")
+
+	return nil
+}
+
+// Transfers a recently minted stToken from the stride-side stake holding address to the return ICA address on the host zone
+func (k Keeper) TransferHoldingToCommunityPoolReturn(ctx sdk.Context, hostZone types.HostZone, coin sdk.Coin) error {
+	memo := ""
+	strideEpochTracker, found := k.GetEpochTracker(ctx, epochstypes.STRIDE_EPOCH)
+	if !found {
+		return errorsmod.Wrapf(types.ErrEpochNotFound, epochstypes.STRIDE_EPOCH)
+	}
+	endEpochTimestamp := uint64(strideEpochTracker.NextEpochStartTime)
+
+	// build and send an IBC message for each coin to transfer all back to the hostZone
+	msg := transfertypes.NewMsgTransfer(
```

Comment:
Just one coin, right?
```suggestion
	// build and send an IBC message for the coin to transfer all back to the hostZone
```

Reply:
This was somehow recommended on the wrong line.